<h1>❤ Heart Attack Classification</h1>



In [9]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn import metrics
from sklearn.preprocessing import RobustScaler


In [10]:
# read data
df = pd.read_csv('heart.csv')

In [11]:
# quick look at data
df.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

In [4]:
df.isnull().sum()

age         0
sex         0
cp          0
trtbps      0
chol        0
fbs         0
restecg     0
thalachh    0
exng        0
oldpeak     0
slp         0
caa         0
thall       0
output      0
dtype: int64

cp = chest pain type
<li>Value 1: typical angina</li>
<li>Value 2: atypical angina</li>
<li>Value 3: non-anginal pain</li>
<li>Value 4: asymptomatic </li>

trtbps = resting blood pressure (in mm Hg)<br/>
chol = cholestoral in mg/dl fetched via BMI sensor<br/>
fbs = (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)<br/>

rest_ecg : resting electrocardiographic results<br/>
<li>Value 0: normal</li>
<li>Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)</li>
<li>Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria</li>

thalach : maximum heart rate achieved<br/>
target : 0= less chance of heart attack 1= more chance of heart attack<br/>

In [12]:
# data preprocessing
df1 = df

cat_cols = ['sex', 'cp', 'restecg', 'fbs', 'exng', 'caa', 'slp', 'thall']
con_cols = ['age', 'trtbps', 'chol', 'thalachh', 'oldpeak']


In [13]:
df1 = pd.get_dummies(df1, columns=cat_cols, drop_first=True)

In [14]:
X = df1.drop(['output'], axis=1)
y = df1[['output']].values

scaler = RobustScaler()
X[con_cols] = scaler.fit_transform(X[con_cols])

model = LogisticRegression()
model.fit(X, y.ravel())



LogisticRegression()

In [16]:
model.score(X, y)


0.8778877887788779

In [17]:
expected = y
predicted = model.predict(X)

print(metrics.classification_report(expected, predicted))
print(metrics.confusion_matrix(expected, predicted))

              precision    recall  f1-score   support

           0       0.89      0.83      0.86       138
           1       0.87      0.92      0.89       165

    accuracy                           0.88       303
   macro avg       0.88      0.87      0.88       303
weighted avg       0.88      0.88      0.88       303

[[115  23]
 [ 14 151]]
